### ISSUE TITLE
Add text to speech for events

### ISSUE THREAD

**Issue by redwolf2** [NPBruce/valkyrie#390]
Having to read all the text from a tablet screen is tireing. Optional TTS could really improve the scenarios, even with a robot voice.

**Comment by LordPyrex** [comment]
Awww, but that's one of my favorite things to do! And I like to make different voices for each character that speaks.

**Comment by redwolf2** [comment]
Entirely optional ;)

**Comment by LordPyrex** [comment]
Yeah, the only problem that I have with text to speech is how dry it comes out sounding. I guess you can just imagine Stephen Hawking narrating your story!

**Comment by redwolf2** [comment]
https://www.youtube.com/watch?v=5bueZoYhUlg&t=58s

The problem is, there are many TTS services and engines available. Most of them sound very similar – quite good but [uncanny](https://en.wikipedia.org/wiki/Uncanny_valley). The best ones are only available via cloud based services you need to pay for, like the [Google Speech API](https://cloud.google.com/speech/), the [Bing Spech API from MS Azure](https://azure.microsoft.com/en-us/pricing/details/cognitive-services/speech-api/) or [Amazon Polly](https://aws.amazon.com/de/polly/). The most promising today is [Google Wavenet](https://deepmind.com/blog/wavenet-generative-model-raw-audio/), but that isn't available to the public yet. The best poor man approach to this would be to just utilize a good offline speech engine, that is available on the specific plattform. I doubt, that there is one, that is available on all plattforms in all required languages. For Windows, there is System.Speech, that performs quite ok.

**Comment by NPBruce** [comment]
That is a reasonable suggestion, perhaps this isn't a Valkyrie thing after all then?  If I support different audio based on language it can be up to the scenario author.

**Comment by antroids** [comment]
Hi guys!
I spent a couple of days and integrated Google TTS.
API key with enabled Cloud TTS required, it would be better to create new for only Valkyrie.
Limitations for WaveNet are: 1 million chars per month for free.
Supported voices: https://cloud.google.com/text-to-speech/docs/voices

The commit is here: https://github.com/antroids/valkyrie/commit/4c81fca4801ab19370b7c2940464f7bbe9b2ec93#diff-a5790b06f63b7c1646f0de34b44fc108377a02fb07aa60b83aaff44deed06398
Feel free to use this code.
